.github/workflows/ci.yaml

```diff
@@ -126,8 +126,8 @@ jobs:
         with:
           provider: microk8s
           channel: "1.28-strict/stable"
-          juju-channel: 3.1/stable
-          bootstrap-options: "--agent-version 3.1.6"
+          juju-channel: 3.5/stable
+          bootstrap-options: "--agent-version 3.5.3"
```

Comment:
**niptick** I would prefer to handle these upgrades with a proper commit, very visible in the history. But in general, for this, I would also prefer to have matrices of juju versions such that we can both tests things that we known are deployed (like 3.1) and the more recent ones

Can we just rollback this change?

Reply:
Happy to remove this change, and add it in a separate PR. I can do that now.
I'm not sure we should be testing against a matrix, given that the directive is to always be recommending deployments against the latest Juju version. 

The reason for this change here, was because the `remove-application` int-tests were failing due to the lack of features/fixes from later Juju versions.

Reply:
Might be more sensible to test PR/publish CI against latest Juju, but have some daily/weekly job against a matrix.

Reply:
Done in [240d981](https://github.com/canonical/kafka-k8s-operator/pull/133/commits/240d981f6b11f4d0b3346fbab45812ba1d6281d0)